app/lib/advanced_text_formatter.rb

```diff
@@ -7,9 +7,10 @@ def initialize(options, &block)
       @format_link = block
     end
 
-    def block_code(code, _language)
+    def block_code(code, language)
+      # Looks wrong, but sets title to language correctly. One downside is, it adds an empty attribute when no lang specified.
       <<~HTML
-        <pre><code>#{ERB::Util.h(code).gsub("\n", '<br/>')}</code></pre>
+        <pre><code title="#{language}">#{ERB::Util.h(code).gsub("\n", '<br/>')}</code></pre>
```

Comment:
I'm not sure `title` is the best option here, but I don't know what would be better. I thought about `lang` but programming languages are explicitly out of scope.

Reply:
I was also considering using a style attribute instead, but as mentioned, using the title attribute has the advantage of showing which code language a code block was written in when hovering it.
I think title is the best option.

Reply:
Hmmh... Now that highlighting requires the class attribute anyway, I think it's actually better to also change title to class. I would still like the element to have a title attribute though.

Reply:
Maybe use a `data-codelang` or something? Do you know if other fediverse software have such a feature, and if so, how they communicate the language being used?

Reply:
Using a data attribute sounds good.
It seems Misskey has syntax highlighting, but looking at their implementation, I can't figure out which attribute they use, if any.

Reply:
Ah, it looks like Misskey just uses a class attribute: `class="language-<langcode>"` on both pre and code elements, which gets then interpreted by their highlighter prism.js, and a "lang" prop internally.
I'm not sure that's what you wanted to know though.